### DISCUSSION TITLE
updateMany should be able to update columns which are foreign keys

### DISCUSSION THREAD

**Discussion by royra** [prisma/prisma#2493] [Q&A] [answered]
Trying the following:

```typescript
await prisma.invitation.updateMany({
  where: { 
    token,
    expiresAt: { gt: now },
  },
  data: {
    status: InvitationStatus.accepted,
    // ERROR: 'acceptedByUserId' does not exist in type 'InvitationUpdateManyMutationInput
    acceptedByUserId: userId,  
  },
})
```

with schema: 
```
model Invitation {
  id               Int       @id @default(autoincrement())
  token            String    @unique
  expiresAt        DateTime
  acceptedByUserId Int?
  acceptedBy       User?     @relation(fields: [acceptedByUserId], references: [id], name: "invitation_acceptedByUserId")
}
```

## Problem

The `InvitationUpdateManyMutationInput` type does not include the column `acceptedByUserId`.

I don't understand why Prisma can't update columns which also happen to be foreign keys.

I know I can use `update` instead, but it doesn't allow me to specify filters which are not unique like `expiresAt: { gt: now }`.

## Suggested solution

Allow Prisma to update foreign key columns.

**Comment by beeplin** [comment]
it is supposed to use `connect` for foreign key updates: 

```js

await prisma.invitation.updateMany({
  where: { 
    token,
    expiresAt: { gt: now },
  },
  data: {
    status: InvitationStatus.accepted,
    // ERROR: 'acceptedByUserId' does not exist in type 'InvitationUpdateManyMutationInput
    acceptedBy: { connect: { id: userId }, // <-- HERE IT GOES
  },
})
```

**Reply by royra** [reply]
I must be missing something, but your suggestion doesn't compile.

Here are the generated types from the `.prisma` directory:

```typescript
export interface InvitationDelegate {
  // ...
    updateMany<T extends InvitationUpdateManyArgs>(
    args: Subset<T, InvitationUpdateManyArgs>
  ): Promise<BatchPayload>
  // ...
}

export type InvitationUpdateManyArgs = {
  data: InvitationUpdateManyMutationInput
  where?: InvitationWhereInput | null
}

export type InvitationUpdateManyMutationInput = {
  id?: number | null
  token?: string | null
  expiresAt?: Date | string | null
}
  
```

As you can see `InvitationUpdateManyMutationInput` doesn't include `acceptedBy`.

**Reply by ryands17** [reply, marked as answer]
Hey @royra :wave: 
Yes you're correct. `updateMany` doesn't allow updating foreign relations, only `update` does at the moment. Could you open a feature request [here](https://github.com/prisma/prisma/issues/new?assignees=&labels=&template=feature_request.md&title=) and add the above use case as well.

For now as a workaround, you would have to use `prisma.raw`.

**Reply by alucardu** [reply]
Will do. @ryands17 one more question. I'm correct in that i cannot store a record directly in a list? Only a reference to a many to many relation record?

**Reply by ryands17** [reply]
Yes you can't store a record directly in a relation, you need to use either `connect` or `create` for relation records.

**Comment by ludralph** [comment]
Hi @royra,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
